packages/flutter/test/widgets/scrollbar_test.dart

```diff
@@ -39,12 +40,20 @@ ScrollbarPainter _buildPainter({
   )..update(scrollMetrics, scrollMetrics.axisDirection);
 }
 
-class _DrawRectOnceCanvas extends Mock implements Canvas { }
+class _DrawRectOnceCanvas extends Fake implements Canvas {
+  List<Rect> rects = <Rect>[];
+
+  @override
+  void drawRect(Rect rect, Paint paint) {
+    rects.add(rect);
+  }
+}
 
 void main() {
   final _DrawRectOnceCanvas testCanvas = _DrawRectOnceCanvas();
   ScrollbarPainter painter;
-  Rect captureRect() => verify(testCanvas.drawRect(captureAny, any)).captured.single as Rect;
+
+  Rect captureRect() => testCanvas.rects.removeLast();
```

Comment:
`.single` instead of `removeLast()` to capture that there's only been one call?

Reply:
I don't think thats how verify works, the verification result is created once.

Reply:
I guess we could clear the state between each test run